pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -0,0 +1,792 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from __future__ import print_function
+from six import StringIO
+from pyutilib.misc import PauseGC
+from pyomo.core.base import (
+    SymbolMap, AlphaNumericTextLabeler, NumericLabeler,
+    Block, Constraint, Expression, Objective, Var, Set, RangeSet, Param,
+    value, minimize, Suffix)
+from pyomo.core.base.component import ComponentData
+import os, sys, subprocess, pipes, math, logging
+
+import pyomo.util.plugin
+from pyomo.opt.base import IOptSolver
+import pyutilib.services
+
+from pyutilib.misc import Options
+
+
+logger = logging.getLogger('pyomo.solvers')
+
+pyutilib.services.register_executable(name="gams")
+ 
+class GAMSSolver(pyomo.util.plugin.Plugin):
+    """A generic interface to GAMS solvers"""
+    pyomo.util.plugin.implements(IOptSolver)
+    pyomo.util.plugin.alias('gams', doc='The GAMS modeling language')
+
+    options = Options()
+
+    def available(self, exception_flag=True):
+        return pyutilib.services.registered_executable("gams") is not None
+ 
+    def solve(self, model, tee=False):
+        """
+        Solver for pyomo models using GAMS. Converts pyomo model
+        to GAMS input file, solves in GAMS, then returns solution
+        values back into pyomo model.
+
+        Uses GAMS Python API. For installation help visit:
+        https://www.gams.com/latest/docs/apis/examples_python/index.html
+
+        self.options:
+            output_filename=None:
+                GAMS model file. If None, will use model name. Kept if given.
+            symbolic_solver_labels=False:
+                Use full Pyomo component names rather than
+                shortened symbols (slower, but useful for debugging).
+            labeler=None:
+                Custom labeler option. Incompatible with symbolic_solver_labels.
+            solve=True:

```

Comment:
Why do you have this argument?  If you are calling a solver's solve method, it should solve.  If you want to just write the GAMS file, you can use `model.write("foo.gms")` or `pyomo convert [...] --output foo.gms`